src/Roslyn.Suppressions/Roslyn.Suppressions.UnitTests/TestHelpers.cs

```diff
@@ -56,4 +59,30 @@ internal static class TestHelper
                 null
                 );
     }
+
+    //This is to make sure normalising the keys done correctly with culture invariant
+    //Lower case of SETTINGSKEY in Turkish is not settingskey but settıngskey
+    //https://en.wikipedia.org/wiki/Dotted_and_dotless_I 
+    public class TemporaryCultureSwitch : IDisposable
+    {
+        private readonly CultureInfo _originalCulture;
+        private readonly CultureInfo _originalUICulture;
+
+        public TemporaryCultureSwitch(CultureInfo cultureInfo)
+        {
+            _originalCulture = Thread.CurrentThread.CurrentCulture;
+            _originalUICulture = Thread.CurrentThread.CurrentUICulture;
+
+            Thread.CurrentThread.CurrentCulture = cultureInfo;
+            Thread.CurrentThread.CurrentUICulture = cultureInfo;
+        }
+
+        public TemporaryCultureSwitch(string cultureName) : this(new CultureInfo(cultureName)) { }
```

Comment:
It looks like this class is always created with code like the following:

```
using (new TemporaryCultureSwitch(new CultureInfo("tr-TR")))
```

i.e. this constructor isn't being used, even though it is simpler.

In practice, we're very unlikely to use any culture other than Turkish in testing, so the code could be even simpler

e.g. a factory method `public static TemporaryCultureScope CreateTurkishScope()`

or even simpler

e.g. the class could be called `TurkishCultureScope` and hard-coded to use Turkish